src/Command/SelfUpdate.php

```diff
@@ -0,0 +1,140 @@
+     */
+    private $client;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('self-update')
+            ->setAliases(['selfupdate'])
+            ->setDescription('Updates manalize to the latest stable version.')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        if ('' === \Phar::running()) {
+            $io->error('Self-update is available only for PHAR installation.');
+
+            return 1;
+        }
+
+        $this->client = new Client();
+
+        try {
+            $latestTag = $this->getLatestTag();
+        } catch (\Exception $e) {
+            $io->error($e->getMessage());
+
+            return 1;
+        }
+
+        if ('v'.$this->getApplication()->getVersion() === $latestTag) {
+            $io->success(sprintf('manalize is already up to date (%s).', $latestTag));
+
+            return 0;
+        }
+
+        try {
+            $this->updateBuild($latestTag);
+        } catch (\PharException $e) {
+            $io->error([sprintf('The latest manalize build is corrupted (%s).', $e->getMessage()), 'Please try to run the command again.']);
+
+            return 1;
+        } catch (\Exception $e) {
+            if (!$e instanceof \UnexpectedValueException && !$e instanceof \PharException) {
+                throw $e;
+            }
+
+            $io->error([sprintf("Unable to download the latest manalize build.\nReason: %s", $e->getMessage()), 'Please try to run the command again.']);
+
+            return 1;
+        }
+
+        $io->success(sprintf('manalize successfully updated to %s.', $latestTag));
+
+        return 0;
+    }
+
+    private function getLatestTag()
+    {
+        $response = $this->client->get(
+            sprintf('https://api.github.com/repos/%s/releases/latest', Application::REPOSITORY_NAME),
+            ['headers' => ['User-Agent' => Application::REPOSITORY_NAME]]
+        );
+
+        if (200 !== $response->getStatusCode()) {
+            throw new \Exception($response->getReasonPhrase());
+        }
+
+        $raw = $response->getBody();
+        $release = json_decode($raw, true);
+
+        if (!$raw || !$release) {
+            throw new \Exception();
+        }
+
+        return $release['tag_name'];
+    }
+
+    private function updateBuild($latestTag)
+    {
+        $localFilename = realpath($_SERVER['argv'][0]) ?: $_SERVER['argv'][0];
+        $tempFilename = basename($localFilename, '.phar').'-tmp.phar';
+
+        $response = $this->client->get(
+            sprintf('https://github.com/%s/releases/download/%s/manalize.phar', Application::REPOSITORY_NAME, $latestTag),
+            ['save_to' => $tempFilename]
+        );
+
+        if (200 !== $response->getStatusCode()) {
+            throw new \Exception($response->getReasonPhrase());
```

Comment:
`\RuntimeException` is more accurate. But I don't know what you want exactly ?
There is no need for a higher level or domain exception I think for now.